modules/components/org.wso2.carbon.appfactory.provisioning.runtime/src/test/java/org/wso2/carbon/appfactory/provisioning/runtime/test/LoggingTest.java

```diff
@@ -52,20 +56,43 @@ public void initialize() throws RuntimeProvisioningException {
      * @throws RuntimeProvisioningException
      */
 
-    @Test(groups = {"org.wso2.carbon.appfactory.provisioning.runtime"}, description = "Kub Logging")
+    @Test(groups = {"org.wso2.carbon.appfactory.provisioning.runtime" }, description = "Kub Logging")
     public void testSnapshotLogs() throws RuntimeProvisioningException, InterruptedException {
+        DeploymentLogs deploymentLogs;
+        Map <String, String> selector = new HashMap<>();
+        selector.put("app" , "My-JAXRS");
+        afKubClient = new KubernetesRuntimeProvisioningService(testUtils.getAppCtx());
 
         //Initially waits until deployment successful
-        Thread.sleep(30000);
-        for (int i = 0; i < 5; i++) {
-            if(afKubClient.getDeploymentStatus(deploymentConfig)){
-                afKubClient.getRuntimeLogs(null);
-            }else {
+        log.info(">> Waiting for deployment to complete : " + deployment.getMetadata().getName());
+        Thread.sleep(60000);
+
+        for (int i = 0; i < 10; i++) {
+            if (testUtils.getPodStatus(namespace, selector)) {
+                deploymentLogs = afKubClient.getRuntimeLogs(null);
+                Map <String, BufferedReader> logs = deploymentLogs.getDeploymentLogs();
+
+                for (Map.Entry<String, BufferedReader> logEntry : logs.entrySet()) {
+                    String record;
+                    StringBuffer buffer = new StringBuffer();
+                    try {
+                        while ((record = logEntry.getValue().readLine())!=null) {
+                             buffer.append(record);
+                        }
+                    } catch (IOException e) {
+                        log.error("Error while reading from the log of deployment : "
+                                + deployment.getMetadata().getName(), e);
+                    }
+                    //Waiting until tomcat starts
+                    Thread.sleep(10000);
```

Comment:
Instead of thread.sleep, can't we use some polling mechanism? 
If not, IMO it is better to add some log info message about the purpose of thread.sleep